paasta_tools/cli/cli.py

```diff
@@ -38,6 +40,33 @@ def error(self, message):
         sys.exit(1)
 
 
+def list_external_commands():
+    p = subprocess.check_output(["/bin/bash", "-c", "compgen -A command paasta-"])
```

Comment:
From my testing it seems like compgen will list functions, which can be exported. This would confuse os.execlp later. Maybe not a big deal.

Looks like the bash argument `-p` (privileged) would prevent that:

```
-p      Turn  on  privileged mode.  In this mode, the $ENV and $BASH_ENV files are not processed, shell
        functions are not inherited from the environment, and the SHELLOPTS, BASHOPTS, CDPATH, and GLO‐
        BIGNORE  variables,  if  they  appear in the environment, are ignored.  If the shell is started
        with the effective user (group) id not equal to the real user (group) id, and the -p option  is
        not supplied, these actions are taken and the effective user id is set to the real user id.  If
        the -p option is supplied at startup, the effective user id is not reset.  Turning this  option
        off causes the effective user and group ids to be set to the real user and group ids.
```

Reply:
👍